src/ResourceManager/PowerBI/Commands.PowerBI/Commands/NewAzureRmPowerBIEmbeddedCapacity.cs

```diff
@@ -0,0 +1,118 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using System;
+using System.Collections;
+using System.Linq;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.PowerBI.Models;
+using Microsoft.Azure.Commands.PowerBI.Properties;
+using Microsoft.Rest.Azure;
+
+namespace Microsoft.Azure.Commands.PowerBI
+{
+    [Cmdlet(VerbsCommon.New, "AzureRmPowerBIEmbeddedCapacity", SupportsShouldProcess = true), 
+        OutputType(typeof(PSPowerBIEmbeddedCapacity))]
+    public class NewPowerBIEmbeddedCapacity : PowerBICmdletBase
+    {
+        [Parameter(
+            ValueFromPipelineByPropertyName = true, 
+            Position = 0, 
+            Mandatory = true,
+            HelpMessage = "Name of resource group under which you want to create the capacity.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true, 
+            Position = 1, 
+            Mandatory = true,
+            HelpMessage = "Name of the capacity to create.")]
+        [ValidateNotNullOrEmpty]
+        public string Name { get; set; }
+
+        [Parameter(
+            ValueFromPipelineByPropertyName = true, 
+            Position = 2, 
+            Mandatory = true,
+            HelpMessage = "Azure region where the capacity should be created.")]
+        [ValidateNotNullOrEmpty]
+        public string Location { get; set; }
```

Comment:
@selasagi please consider adding the `LocationCompleter` for your creation scenarios that have a `Location` parameter: https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#location-completer

This will allow users to tab through valid locations in which they can create the given resource (by using the resource type to determine valid locations)

Reply:
added. thanks